pkg/providers/v1/providers.go

```diff
@@ -117,6 +118,22 @@ type GitHub interface {
 	AddAuthToPushOptions(ctx context.Context, options *git.PushOptions) error
 }
 
+// ImageLister is the interface for listing images
```

Comment:
What sort of providers would be able to list images, but not list tags?

Reply:
none, these are usually used in conjunction. However, the OCI spec does not have an easy way to list images, we normally have to resort to registry-specific functions for that. For more context, see the original POC PR https://github.com/stacklok/minder/pull/2765

Reply:
So, these are basically the trait implementations. Different registries will be able to use the OCI implementation and just list tags.